src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -210,13 +212,13 @@ internal static async Task<DownloadDependencyResolutionResult[]> DownloadDepende
         {
             telemetryActivity.StartIntervalMeasure();
 
-            var downloadDependencyResolutionTasks = new List<Task<DownloadDependencyResolutionResult>>();
-            foreach (var targetFrameworkInformation in packageSpec.TargetFrameworks)
+            List<Task<DownloadDependencyResolutionResult>> downloadDependencyResolutionTasks = new();
```

Comment:
```suggestion
            List<Task<DownloadDependencyResolutionResult>> downloadDependencyResolutionTasks = new(packageSpec.TargetFrameworks.Count);
```

Minimal gains, good practice.

Reply:
I've updated the method to used properly sized lists and NoAllocEnumerate